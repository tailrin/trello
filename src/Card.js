import React from 'react';
import './Card.css';

export default function Card(props){
    if(Object.entries(props).length === 0){
        props = {
            title: 'this was left blank',
            content: 'this was left blank'
        }
    }

    return(
        <div className="Card">
            <button type="button">delete</button>
            <h3>{props.title}</h3>
            <p>{props.content}</p>
        </div>
    );
}