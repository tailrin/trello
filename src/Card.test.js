import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from './Card.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Card title="This was left blank" content="This was left blank"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });