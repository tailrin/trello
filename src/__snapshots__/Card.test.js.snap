// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    This was left blank
  </h3>
  <p>
    This was left blank
  </p>
</div>
`;
